Sources/XcodeGenKit/NSCacheExtensions.swift

```diff
@@ -0,0 +1,37 @@
+import XcodeProj
+import Foundation
+
+enum Cached<T> {
+    case cached(T)
+    case nothing
+
+    var value: T? {
+        switch self {
+        case let .cached(value): return value
+        case .nothing: return nil
+        }
+    }
+}
+
+final class CacheContainer {
```

Comment:
Why do we need this container? Why not to store plain Cached<BuildSetting>?

Reply:
And minor - do we actually need Cached enum? It just duplicates Optional.

Reply:
> Why do we need this container? Why not to store plain Cached?
NSCache works only with ref types

> And minor - do we actually need Cached enum? It just duplicates Optional.
It was already there, let's keep it. 
IMO this adds more clarity at the point of use.